tensilelite/Tensile/KernelWriter.py

```diff
@@ -4028,35 +4028,38 @@ def readWriteVectors(mat, vw, kernel):
     self.states.a.numVgprLocalWriteAddrTailLoop = 0 if not (kernel["DirectToLdsA"] and kernel["NonDTLTailLoopA"]) else 1 * self.states.rpla
     self.states.b.numVgprLocalWriteAddrTailLoop = 0 if not (kernel["DirectToLdsB"] and kernel["NonDTLTailLoopB"]) else 1 * self.states.rpla
 
-    # TODO: Refactor vgpr multiplier calculation
-    # based on comments here: https://github.com/ROCm/hipBLASLt/pull/2061/files#r2085714139
-    if self.states.archCaps["DeviceLDS"] > 65536 and not kernel["StoreSwapAddr"]:
-      need128K = kernel["LdsOffsetA_Blk"]>=131072 and kernel["ExpandPointerSwap"] and not kernel["1LDSBuffer"]
-      need64K = kernel["LdsOffsetA_Blk"]>=65536 and kernel["ExpandPointerSwap"] and not kernel["1LDSBuffer"]
+    numVgprMultiplierA = 1
+    numVgprMultiplierB = 1
+    numVgprMultiplierMetadata = 1
+    if self.states.archCaps["DeviceLDS"] > 65536:
+      need128K = kernel["LdsOffsetA_Blk"]>=131072 and kernel["ExpandPointerSwap"] and not kernel["1LDSBuffer"] and not kernel["StoreSwapAddr"]
```

Comment:
I still prefer to use division instead of checking >128K, >64K individually.

Like this.
apply_LdsOffsetA_Blk = kernel["ExpandPointerSwap"] and not kernel["1LDSBuffer"] and not kernel["StoreSwapAddr"]
if apply_LdsOffsetA_Blk:
  numVgprMultiplierA = kernel["LdsOffsetA_Blk"]//maxLDSConstOffset + 1

numVgprMultiplierA = max(numVgprMultiplierA , kernel["LdsNumElementsAlignedA"]//maxLDSConstOffset  + 1)

Reply:
Good suggestion, updated.